notebooks/_06-Architecture.md

```diff
@@ -0,0 +1,548 @@
+---
+jupyter:
+  jupytext:
+    formats: ipynb,md
+    text_representation:
+      extension: .md
+      format_name: markdown
+      format_version: '1.2'
+      jupytext_version: 1.4.1
+  kernelspec:
+    display_name: Python 3
+    language: python
+    name: python3
+---
+
+# Architecture
+
+
+## 1. At a “high” level, how will the project deliver each of the needed requirements?
+- **Try to think in terms of some manageable (e.g. 10-20) steps**
+- **List needed intermediate products (e.g. pseudo-code, unit tests, etc.)**
+- **Try to be concise but comprehensive?**
+
+The project is split into three main tasks:
+
+1. The first task is focused on a selection-on-observables problem:
+
+For a causal graph with independent covariates:
+ 1. Ingest data
+ 2. Manipulate data as needed, the output of this step will be a cleaned dataset
+ 3. Specify a choice model based on the data at hand, the specification should follow required conventions by third party dependencies
+ 4. The cleaned dataset will be used in a function that estimated the distributions for variables of interest, this function will produce a parameter dictionary containing the type of variables, the best fitted distribution, and that distribution parameters.
+ 5. The resulting parameter dictionary from the previous step will be used to simulate needed data
```

Comment:
For independent variables only. I didn't see anywhere here that you'll be using the causal graph structure for simulating variables that have parents